storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManager.java

```diff
@@ -567,9 +567,9 @@ public TopicBasedRemoteLogMetadataManagerConfig config() {
     public void close() throws IOException {
         // Close all the resources.
         log.info("Closing topic-based RLMM resources");
-        if (closing.compareAndSet(false, true)) {
-            lock.writeLock().lock();
-            try {
+        lock.writeLock().lock();
```

Comment:
Why do we need the `writeLock` in the close method? Can we avoid taking the lock in the close method? 

We already have AtomicBoolean `closing` to inform the callers that this instance is about to be closed.

Reply:
yes, you are right. We can avoid taking lock in close method.

After setting closing to true there are 2 cases: 

1. initializationThread is null -> In this case, we won't wait for this to complete. Because closing = true will prevent initializationThread from creating any resources.
2. initializationThread is not null -> In this case we will wait for completion of initializationThread.